src/sql-parser/src/ast/defs/name.rs

```diff
@@ -49,18 +50,26 @@ impl Ident {
     ///
     /// let too_long_id = Ident::new(too_long);
     /// assert!(too_long_id.is_err());
+    ///
+    /// let invalid_name_dot = Ident::new(".");
+    /// assert!(invalid_name_dot.is_err());
+    ///
+    /// let invalid_name_dot_dot = Ident::new("..");
+    /// assert!(invalid_name_dot_dot.is_err());
     /// ```
     ///
-    pub fn new<S>(value: S) -> Result<Self, IdentError>
+    pub fn new<S>(s: S) -> Result<Self, IdentError>
     where
-        S: Into<String>,
+        IdentString: TryFrom<S>,
```

Comment:
types that implement `TryFrom` automatically implement `TryInto` in the other direction! Slight preference in changing this to `S: TryInto<IdentString>` since IMO it's easier to read

Reply:
Done.